File: app/src/main/java/com/sachtech/datingapp/ui/profile/ProfileItemFragment.java
package com.sachtech.datingapp.ui.profile;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.TextView;

import androidx.fragment.app.DialogFragment;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.RecyclerView.LayoutManager;

import com.google.android.flexbox.FlexDirection;
import com.google.android.flexbox.FlexboxLayoutManager;
import com.google.android.flexbox.JustifyContent;
import com.sachtech.datingapp.R;
import com.sachtech.datingapp.R.id;
import com.sachtech.datingapp.ui.home.adapter.ProfileItemListAdapter;
import com.sachtech.datingapp.ui.profile.listener.EditInformation;
import com.sachtech.datingapp.ui.profile.listener.OnItemSelected;
import com.sachtech.datingapp.utils.ProfileItem;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.Arrays;


public final class ProfileItemFragment extends DialogFragment implements OnItemSelected {
   @Nullable
   private TextView textView;
   @Nullable
   private EditInformation onEditInformation;
   @Nullable
   private ArrayList list;
   @Nullable
   private String selectedItem;
   @Nullable
   private ProfileItemListAdapter adapter;
   @Nullable
   private String selectedType;
   @NotNull
   private String[] booleanArray = new String[]{"Yes", "No"};
   @NotNull
   private String item = "";


   @NotNull
   public final String getItem() {
      return this.item;
   }

   public final void setItem(@NotNull String var1) {
      this.item = var1;
   }

   public void onViewCreated(@NotNull View view, @Nullable Bundle savedInstanceState) {
      super.onViewCreated(view, savedInstanceState);
      getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
      Bundle bundle = this.getArguments();
      this.selectedItem = bundle.getString("item_name");
      this.selectedType =  bundle.getString("item_type");
      TextView itemName = view.findViewById(id.item_name);
      itemName.setText(selectedItem);
      this.list = new ArrayList();
      RecyclerView rv = view.findViewById(id.profile_item_rv);

      FlexboxLayoutManager layoutManager = new FlexboxLayoutManager(this.getContext());
      layoutManager.setFlexDirection(FlexDirection.ROW);
      layoutManager.setJustifyContent(JustifyContent.CENTER);
      rv.setLayoutManager((LayoutManager)layoutManager);
       adapter =new ProfileItemListAdapter(getContext(), list, selectedItem, selectedType,this);

      list=new ArrayList();
     // ArrayList var10004 = this.list;
     if(this.selectedItem==ProfileItem.INSTANCE.getEDUCATION())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.education)));
     }
     else if(selectedItem== ProfileItem.INSTANCE.getPROFESSION())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.profession)));
     }
     else if(selectedItem== ProfileItem.INSTANCE.getEYECOLOR())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.eyeColor)));
     }
     else if(selectedItem== ProfileItem.INSTANCE.getHAIRCOLOR())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.hairColor)));
     }
     else if(selectedItem==ProfileItem.INSTANCE.getCOLORCOMPLEXION())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.colorComplexion)));

     } else if(selectedItem==ProfileItem.INSTANCE.getQURAN())
     {
        list.addAll(Arrays.asList(booleanArray));
     }
     else if(selectedItem==ProfileItem.INSTANCE.getPRAYS())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.prays)));
     }
     else if(selectedItem==ProfileItem.INSTANCE.getHALAL())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.halal)));
      }
     else if(selectedItem==ProfileItem.INSTANCE.getWEAR())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.wear)));
     }
     else if(selectedItem==ProfileItem.INSTANCE.getLIFEAFTERMARRIAGE())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.after_marriage_plans)));
     }
     else if(selectedItem==ProfileItem.INSTANCE.getLIFEAFTERMARRIAGE())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.after_marriage_plans)));
     }
     else if(selectedItem==ProfileItem.INSTANCE.getMARRIAGEPLANS())
     {
        list.addAll(Arrays.asList(getResources().getStringArray(R.array.marriage_plans)));
     }
     else if(selectedItem==ProfileItem.INSTANCE.getREVERT())
     {
        list.addAll(Arrays.asList(booleanArray));
     }
     else if(selectedItem==ProfileItem.INSTANCE.getBEARD())
     {
        list.addAll(Arrays.asList(booleanArray));
     }
     TextView tv=view.findViewById(id.profile_item_save);
     tv.setOnClickListener(new OnClickListener() {
        @Override
        public void onClick(View v) {
            onEditInformation.editinformation(item,textView);
            dismiss();
           }
     });

   }

   public final void infoSelectListener(@NotNull EditInformation onEditInformation, @NotNull TextView textView) {
      this.onEditInformation = onEditInformation;
      this.textView = textView;
   }

   public void selectedItem(@NotNull String item) {
      this.item = item;
   }
}

File: app/src/main/java/com/sachtech/datingapp/ui/profile/ProfileItemFragment.kt
/*
package com.sachtech.datingapp.ui.profile

import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.google.android.flexbox.FlexDirection
import com.google.android.flexbox.FlexboxLayoutManager
import com.google.android.flexbox.JustifyContent
import com.sachtech.datingapp.R
import com.sachtech.datingapp.ui.home.adapter.ProfileItemListAdapter
import com.sachtech.datingapp.ui.home.adapter.listener.OnItemSelected
import com.sachtech.datingapp.ui.profile.listener.EditInformation
import com.sachtech.datingapp.utils.ProfileItem
import gurtek.mrgurtekbase.KotlinBaseDialogFragment
import kotlinx.android.synthetic.main.fragment_profile_item.*

class ProfileItemFragment : KotlinBaseDialogFragment<String>(R.layout.fragment_profile_item),
    OnItemSelected {
    override fun adjustdisplay(): Boolean {
        return true
    }


    var textView: TextView? = null
    var onEditInformation: EditInformation? = null

    var list: ArrayList<String>? = null
    var selectedItem: String? = null
    var adapter: ProfileItemListAdapter? = null
    var selectedType: String? = null
    var booleanArray = arrayOf("Yes", "No")
    var item=""
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        dialog?.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))

        selectedItem = arguments?.getString("item_name")
        selectedType = arguments?.getString("item_type")

        item_name.text = selectedItem
        list = ArrayList()
        val layoutManager = FlexboxLayoutManager(context)
        layoutManager.flexDirection = FlexDirection.ROW
        layoutManager.justifyContent = JustifyContent.CENTER
        profile_item_rv.layoutManager = layoutManager
        adapter = ProfileItemListAdapter(context!!, list!!, selectedItem, selectedType,this)
        profile_item_rv.adapter = adapter

        when (selectedItem) {
            ProfileItem.EDUCATION -> list?.addAll(resources.getStringArray(R.array.education))
            ProfileItem.PROFESSION -> list?.addAll(resources.getStringArray(R.array.profession))
            ProfileItem.EYECOLOR -> list?.addAll(resources.getStringArray(R.array.eyeColor))
            ProfileItem.HAIRCOLOR -> list?.addAll(resources.getStringArray(R.array.hairColor))
            ProfileItem.COLORCOMPLEXION -> list?.addAll(resources.getStringArray(R.array.colorComplexion).toList())
            ProfileItem.QURAN -> list?.addAll(booleanArray)
            ProfileItem.PRAYS -> list?.addAll(resources.getStringArray(R.array.prays))
            ProfileItem.HALAL -> list?.addAll(resources.getStringArray(R.array.halal))
            ProfileItem.WEAR -> list?.addAll(resources.getStringArray(R.array.wear))
            ProfileItem.LIFEAFTERMARRIAGE -> list?.addAll(resources.getStringArray(R.array.after_marriage_plans))
            ProfileItem.MARRIAGEPLANS -> list?.addAll(resources.getStringArray(R.array.marriage_plans))
            ProfileItem.REVERT -> list?.addAll(booleanArray)
            ProfileItem.BEARD -> list?.addAll(booleanArray)
        }

        profile_item_save.setOnClickListener {
            onEditInformation?.editinformation(item ?: "", textView!!)
            dismiss()
        }
    }

    fun infoSelectListener(onEditInformation: EditInformation, textView: TextView) {
        this.onEditInformation = onEditInformation
        this.textView = textView
    }

    override fun selectedItem(item: String) {
        this.item=item
    }


}
*/
